kernel/src/main/scala/cats/kernel/Comparison.scala

```diff
@@ -0,0 +1,19 @@
+package cats.kernel
+
+/** ADT encoding the possible results of a comparison */
+sealed abstract class Comparison extends Product with Serializable {
+  /** The signum of this comparison */
+  def toInt: Int = this match {
+    case Comparison.GreaterThan => 1
+    case Comparison.EqualTo     => 0
+    case Comparison.LessThan    => -1
+  }
```

Comment:
Yeah I think having `Comparison` instances "know" their integer value (either through fields or through methods on the subtype) would be better than using pattern matching in `.toInt`.